apps/docs/build.md

```diff
@@ -0,0 +1,218 @@
+   visit those pages multiple times per browsing session, and across multiple
+   days, since many of them are students and teachers in a regularly scheduled
+   class.
+
+6. Visitors to the marketing pages come from all over the world and have all
+   different sorts of internet connections (both in bandwidth and
+   latency). Chances are pretty good that they have not visited these pages
+   recently if ever, though once they do visit them, they typically visit many
+   of them in a row.
+
+7. We deploy a new version of the site, which always includes javascript
+   changes, 5 days a week, every week.
+
+With these constraints in mind, here are the high level guidelines we shoot for
+when configuring our bundles:
+
+- _DO_ try to factor out code used on more than one page into a shared bundle to
+  improve client-side cache efficiency.
+- _DO NOT_ try to factor out _all_ code used on more than one page into a single
+  shared bundle to reduce the total number of bytes that must be downloaded for
+  any given page on a cache miss.
+- _DO_ try to factor out code that rarely changes into a single bundle that can
+  be cached for a long time on the client across multiple production
+  deployments.
+- _DO_ try to factor out code that is used on multiple pages which are often
+  visited in succession.
+- _DO_ try to have a small number of bundles for any given page (<= 5)
+
+Currently, the outcome of applying these guidelines as best we can is the
+following:
+
+- `minimal.js` - this bundle includes code that is used pretty much everywhere
+  on the site. It will get loaded on every page and (hopefully) doesn't change
+  that often.
+
+- `code-studio-common.js` - this bundle includes code that is used on all
+  studio.code.org pages, but which isn't already in `minimal.js`. So any page
+  that needs `code-studio-common.js`, also needs `minimal.js` to work.
+
+- `common.js` - this bundle includes code that is used across all of the big
+  apps (gamelab, applab, etc.) but which isn't already in `minimal.js` or
+  `code-studio-common.js`.
+
+Then there are a bunch of much smaller bundles which serve as the "entry points"
+for various pages. They contain code which isn't used anywhere else and is
+really only for one specific page. So in theory, even the heaviest pages should
+only require downloading 4 different javascript files.
+
+### When should I create a new bundle?
+
+The only time you should create a new bundle is when you are creating a new
+rails page at a new url. If you are creating lots and lots of these pages, then
+it _might_ be reasonable to create a new common chunk bundle, but this should be
+considered carefully before being implemented.
+
+### How do I create a new bundle?
+
+Creating a new bundle is easy, but you need to think about what that bundle's
+dependencies will be. Here are the different options:
+
+1. I'm creating a new "app" like applab/gamelab/etc.
```

Comment:
More precisely a new "code studio app", possibly useful to differentiate from a "dashboard app" such as pixelation or multi/match